packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -248,8 +248,6 @@ export class BaseInvocationScope<TReturn = any> {
     const txCost = await this.getTransactionCost();
     const { gasUsed, missingContractIds, outputVariables, maxFee } = txCost;
     this.setDefaultTxParams(transactionRequest, gasUsed, maxFee);
-    // Clean coin inputs before add new coins to the request
-    transactionRequest.inputs = transactionRequest.inputs.filter((i) => i.type !== InputType.Coin);
```

Comment:
To successfully close #3750, we also need to stop `clone()` in these places, no?

https://github.com/FuelLabs/fuels-ts/blob/f41887f754febbb4469d990e1cdf0d74fc9d9160/packages/program/src/functions/base-invocation-scope.ts#L230

https://github.com/FuelLabs/fuels-ts/blob/f41887f754febbb4469d990e1cdf0d74fc9d9160/packages/program/src/functions/base-invocation-scope.ts#L246

Reply:
I believe cloning the TX request was not the source of the problem. The clone occurs only after `.call` is invoked. By that point, all custom modifications to the TX request have already been applied.

Additionally, there's another cloning step before calling `getTransactionCost`, but this is only to ensure that the request remains unmodified during estimation.

The real issue was that all input coins were being removed during `.call`, causing any manually added UTXOs to be discarded.